src/slider/slider.tsx

```diff
@@ -0,0 +1,285 @@
+    if (nextValues[0] > nextValues[1]) {
+      nextValues.reverse();
+      setDraggedIndex(prevState => (prevState === 0 ? 1 : 0));
+    }
+    handleValueChange(nextValues);
+  }, [validValues, draggedIndex, max, min, validStep, handleValueChange]);
+
+  const handleKeyDown = useCallback(({key, currentTarget}: KeyboardEvent) => {
+    const nextValues = [...validValues];
+    const index = Number(currentTarget.getAttribute('data-index'));
+    switch (key) {
+      case 'ArrowLeft':
+      case 'ArrowDown':
+        nextValues[index] = Math.max(min, validValues[index] - validStep);
+        break;
+      case 'ArrowRight':
+      case 'ArrowUp':
+        nextValues[index] = Math.min(max, validValues[index] + validStep);
+        break;
+      case 'Home':
+        nextValues[index] = min;
+        break;
+      case 'End':
+        nextValues[index] = max;
+        break;
+      default:
+        return;
+    }
+    if (nextValues[0] > nextValues[1]) {
+      const previousValue = nextValues[index];
+      nextValues.reverse();
+      const thumb: HTMLButtonElement | null | undefined = ref.current?.querySelector(
+        `[role="slider"][data-index="${nextValues.indexOf(previousValue)}"]`
+      );
+      thumb?.focus();
+    }
+    handleValueChange(nextValues);
+  }, [validValues, handleValueChange, min, validStep, max]);
+
+  useEffect(() => {
+    if (disabled) {
+      return undefined;
+    }
+
+    if (isDragging && !previouslyDragged.current) {
+      window.addEventListener('mousemove', handleMouseMove);
+      window.addEventListener('mouseup', handleMouseUp);
+    } else if (!isDragging && previouslyDragged.current) {
+      window.removeEventListener('mousemove', handleMouseMove);
+      window.removeEventListener('mouseup', handleMouseUp);
+    }
+    return () => {
+      window.removeEventListener('mousemove', handleMouseMove);
+      window.removeEventListener('mouseup', handleMouseUp);
+    };
+  }, [isDragging, handleMouseMove, handleMouseUp, disabled]);
+
+  return (
+    <div
+      ref={ref}
+      role="button"
+      className={classNames(styles.slider, className, {
+        [styles.disabled]: disabled,
+        [styles.marked]: !!marks || showTag
+      })}
+      tabIndex={-1}
+      onMouseDown={handleMouseDown}
+    >
+      <div className={classNames(styles.rail, {
+        [styles.rounded]: !showTicks,
+        [styles.disabled]: disabled
+      })}
+      />
+      <div
+        style={{
+          left: `${trackStart}%`,
+          width: `${trackLength}%`
+        }}
+        className={classNames(styles.track, {
+          [styles.rounded]: !showTicks,
+          [styles.disabled]: disabled
+        })}
+      />
+      {validValues.map((numValue, index) => {
+        const percent = toPercent(numValue, min, max);
+        return (
+          // eslint-disable-next-line react/no-array-index-key
+          <Fragment key={index}>
```

Comment:
Maybe `numValue` is better option for key?

Reply:
Yes, I thought about it, too, but it creates an ambiguous situation for a 'range' slider when we have two thumbs pointing at the same value and end up having two duplicate keys.